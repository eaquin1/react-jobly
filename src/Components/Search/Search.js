import React, {useState} from "react"

function Search({searchFor}) {

    const [search, setSearch] = useState('');

    const handleChange = (e) => {
        setSearch( e.target.value)
    }

    const handleSubmit = (e) => {
        e.preventDefault()
        searchFor(search)
    }
return (
    <div>
        <form onSubmit={handleSubmit}>
            <input
                name="search"
                placeholder="Enter a search term"
                value={search} 
                onChange={handleChange}
            />
            <button>Submit</button>

        </form>
    </div>
)
}


export default Search