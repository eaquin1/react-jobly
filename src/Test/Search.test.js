import React from 'react';
import { render } from '@testing-library/react';
import Search from '../Components/Search/Search';

describe('Testing the Search component', () => {
  it('renders without crashing', () => {
    render(<Search />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });
});