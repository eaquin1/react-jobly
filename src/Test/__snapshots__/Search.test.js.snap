// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Search component matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="margin-left: 40px;"
  >
    <form
      class="form-inline"
    >
      <input
        class="form-control"
        name="search"
        placeholder="Enter a search term"
        type="text"
        value=""
      />
      <button
        class="btn btn-secondary"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
